### ISSUE TITLE
Adding native/custom PiP buttons to popular sites

### ISSUE THREAD

**Issue by arnoappenzeller** [arnoappenzeller/PiPifier#12]
2d783e04d24d9e6e28d7de48bd65b650c0bbfb54 adds the possibility to enable custom PiP in the video player of popular pages. This option is enabled by default since I find the inline PiP much more useful.

One major advantage is that this doesn't require any other interaction that you would need for autoplaying video players.

At the moment this feature only supports Youtube.

Here is an updating list of video players that are supported/deprecated(because they added their own button)/wished:

* Supported:
    - Youtube
* Deprecated:
* Wishlist:
    - Netflix

If you have any player you want to see supported don't hesitate and I add it to the wishlist

**Comment by RvMouche** [comment]
A native player in videoJS players would be neat

**Comment by arnoappenzeller** [comment]
Should be possible but there needs to be a clever way to detect the videoJS players because currently the native placement works via the domain.

**Comment by arnoappenzeller** [comment]
365c03b2b9da6624aa4dded92a508345304b96e9 adds basic support for video.js players.

But I'm lacking a bit of real world examples. @RvMouche do you have some maybe?

**Comment by kwokaaron** [comment]
Slightly off-topic, but wanted to check the feasibility of this first. I was wondering with the addition of the custom PiP buttons in the video players on popular pages, is it possible to provide an option to disable this feature as well in the extension settings?

**Comment by arnoappenzeller** [comment]
The main app has an option to disable it so if a problem would occur this would be a workaround till a fix
![bildschirmfoto 2016-10-18 um 20 53 26](https://cloud.githubusercontent.com/assets/4069713/19491918/0511cffe-9575-11e6-92ad-5f5fa1afbd00.png)

**Comment by kwokaaron** [comment]
Ah sorry about that, I was only checking to see if there was an option to do this in the Safari preference 'Extensions' tab. Thanks!

**Comment by arnoappenzeller** [comment]
Don't worry I wanted to add it there in the first place but it seems not possible anymore with the new Safari Extension API

**Comment by kwokaaron** [comment]
Got it. Got slightly confused since my old extensions previously installed in Safari still shows this.

**Comment by kwokaaron** [comment]
There seems to be a bug where it reappears. It seems to work on the very first page load, but refreshing the youtube video page seems to show the button again.

**Comment by arnoappenzeller** [comment]
Did you disable it and then it reappears?

**Comment by kwokaaron** [comment]
Yes. I've even tried uninstalling it and deleting all the containers.

**Comment by arnoappenzeller** [comment]
This should be considered as a bug and is noted in #15

**Comment by akeonline** [comment]
Hi, am using JW Player (flash) on streamtvbox website, but the button doesn't seem to appear to be able to click it, I've tried pausing and playing, nothing seems to work? Any help would be greatly appreciated, thanks.

**Comment by arnoappenzeller** [comment]
Flash is not supported by PiP mode in Sierra so this is not possible. Also this is not the right issue for it so I'll clean this up in a bit

**Comment by WillyH93** [comment]
Maybe outdated, but unfortunately some websites still use Silverlight (by Microsoft). I'm just wondering if it would be possible to see PiPifier supporting those videos in some future update?

**Comment by arnoappenzeller** [comment]
@WillyH93 unfortunately PiP on Sierra supports only HTML5 videos. So Silverlight can't be supported at the moment

**Comment by JoniVR** [comment]
It would be great if it would support Udemy videos, I tried it and didn't work, Udemy somehow blocks right clicking, but if you go into the source of the webpage, open the video link directly in a new tab, you can click picture-in-picture, so it is possible.. I was actually looking to write some little script myself until I stumbled across this, great work so far 👍🏻

**Comment by WilliamBlais** [comment]
Would be awesome to add Udemy.

**Comment by JoniVR** [comment]
@WilliamBlais  As far as I'm aware, it already supports Udemy, for me the button is on the bottom left next to play
![screenshot](https://cloud.githubusercontent.com/assets/7591717/24288190/ba381974-107c-11e7-8cd3-26983bb1b258.jpeg)

**Comment by tonycoco** [comment]
What about Facebook?

**Comment by adavidzh** [comment]
I was wondering about support for [RTP](http://www.rtp.pt/play/direto/rtpmadeira) since that uses the HTML5 [Radiant Media Player](https://www.radiantmediaplayer.com/).

**Comment by olliebaum** [comment]
So, it's not strictly a video player... but would it be possible to add support for **Tumblr TV**. I like to have GIFs playing in the background while I work. Would be neat to have it in PiP.

**Comment by floriandierickx** [comment]
To add to the wishlist: [**THEOPLayer**](https://www.theoplayer.com) would be nice. It is used by vualto, .. to stream NBC, CNN, Vrt (belgian public broadcasting), ...
